spec/rspec/rails/matchers/active_job_spec.rb

```diff
@@ -196,6 +196,19 @@ def self.name; "LoggingJob"; end
       }.to have_enqueued_job.at(date)
     end
 
+    it "has an enqueued job when providing at of :no_wait and there is no wait" do
+      expect {
+        hello_job.perform_later
+      }.to have_enqueued_job.at(:no_wait)
+    end
+
+    it "does not have an enqueued job when providing at of :no_wait and there is a wait" do
+      date = Date.tomorrow.noon
+      expect {
+        hello_job.set(:wait_until => date).perform_later
+      }.to_not have_enqueued_job.at(:no_wait)
+    end
```

Comment:
There's another scenario here, not setting at `at` all and there being a wait

Reply:
good callouts 👍